lib/vector-config/src/stdlib.rs

```diff
@@ -21,51 +22,53 @@ use crate::{
 };
 
 // Unit type.
-impl<'de> Configurable<'de> for () {
-    fn generate_schema(_: &mut SchemaGenerator, _: Metadata<'de, Self>) -> SchemaObject {
+impl Configurable for () {
+    fn generate_schema(_: &mut SchemaGenerator, _: Metadata<Self>) -> SchemaObject {
         panic!("unit fields are not supported and should never be used in `Configurable` types");
     }
 }
 
 // Null and boolean.
-impl<'de, T> Configurable<'de> for Option<T>
+impl<T> Configurable for Option<T>
 where
-    T: Configurable<'de>,
+    T: Configurable + Serialize,
 {
     fn is_optional() -> bool {
         true
     }
 
-    fn metadata() -> Metadata<'de, Self> {
+    fn metadata() -> Metadata<Self> {
         // We clone the default metadata of the wrapped type because otherwise this "level" of the schema would
         // effective sever the link between things like the description of `T` itself and what we show for a field of
         // type `Option<T>`.
         //
         // To wit, this allows callers to use `#[configurable(derived)]` on a field of `Option<T>` so long as `T` has a
         // description, and both the optional field and the schema for `T` will get the description... but the
         // description for the optional field can still be overridden independently, etc.
-        T::metadata().map_default_value(|default| Some(default))
+        T::metadata().convert()
```

Comment:
I changed this because what this was doing was always setting a default value for an optional field based on the default value of its contained `T`.

That isn't what the default of an `Option<T>` would be if you just did `Default::default()`, so the previous code was not correct in that case. Callers can still specify a default for an optional field at the field-level if need be, but we shouldn't add one by default via `T`.